main.tf

```diff
@@ -0,0 +1,65 @@
+resource "azurerm_resource_group" "rg" {
+  name     = "incident-backstage-rg"
+  location = "UK South"
+}
+
+resource "random_password" "password" {
+  length  = 16
+  special = true
+  upper   = true
+  lower   = true
+  number  = true
+}
+
+data "azurerm_key_vault" "ptl" {
+  name                = "cft${var.env}-intsvc"
+  resource_group_name = "core-infra-intsvc-rg"
+}
+
+resource "azurerm_key_vault_secret" "backstage-db-secret" {
+  name         = "backstage-db-password"
+  value        = random_password.password.result
+  key_vault_id = data.azurerm_key_vault.ptl.id
+}
+
+variable "env" {
+}
+
+resource "azurerm_postgresql_server" "db" {
```

Comment:
It will be good if we can tune our postgres module and use module here instead of resources.